File: julia/bob/bob.jl
using Match

shouting(stimulus) = uppercase(stimulus) == stimulus && any(isletter,stimulus)
question(stimulus) = endswith(stimulus,"?")
silence(stimulus) = isempty(stimulus) 

function bob(stimulus::AbstractString) 
  @match stimulus |> strip begin
    s, if shouting(s) && question(s) end => "Calm down, I know what I'm doing!"
    s, if shouting(s) end => "Whoa, chill out!"
    s, if question(s) end => "Sure."
    s, if silence(s) end => "Fine. Be that way!"
    _ => "Whatever."
  end
end

File: fsharp/bob/Bob.fs
module Bob

let response (input: string) =
    let shouting (input: string) =
        input = input.ToUpper()
        && input <> input.ToLower()

    let question (input: string) = (input.Trim() |> Seq.last) = '?'
    let forceFullQuestion input = shouting input && question input
    let silence (input: string) = input.Trim() = ""

    match input with
    | input when silence input -> "Fine. Be that way!"
    | input when forceFullQuestion input -> "Calm down, I know what I'm doing!"
    | input when shouting input -> "Whoa, chill out!"
    | input when question input -> "Sure."
    | _ -> "Whatever."
